debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -314,10 +312,7 @@ private void addSchemaEvent(RelationalSnapshotContext<MySqlPartition, MySqlOffse
 
         List<SchemaChangeEvent> schemaChangeEvents = databaseSchema.parseSnapshotDdl(snapshotContext.partition, ddl, database,
                 snapshotContext.offset, clock.currentTimeAsInstant());
-        List<SchemaChangeEvent> missingSchemaChangeEvents = schemaChangeEvents.stream()
-                .filter(not(schemaEvents::contains))
-                .collect(Collectors.toList());
-        schemaEvents.addAll(missingSchemaChangeEvents);
+        schemaEvents.addAll(new HashSet<>(schemaChangeEvents));
```

Comment:
If there are multiple events we'd lose the order in which they are applied.

Reply:
Changed it with `LinkedHashSet` that maintain the same performance but with ordering.